From: Lucas Kanashiro <kanashiro@ubuntu.com>
Date: Mon, 10 Jun 2024 20:22:25 -0300
Subject: Do not pass -m32 to arm 32 bits
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

The -m32 flag in already 32 bits architecture is generating the
following error:

arm-linux-gnueabi-gcc: error: unrecognized command-line option ‘-m32’; did you mean ‘-mbe32’?
---
 tool/m4/ruby_default_arch.m4 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tool/m4/ruby_default_arch.m4 b/tool/m4/ruby_default_arch.m4
index 30f5f9c..4362b40 100644
--- a/tool/m4/ruby_default_arch.m4
+++ b/tool/m4/ruby_default_arch.m4
@@ -4,7 +4,6 @@ AC_DEFUN([RUBY_DEFAULT_ARCH], [
 AC_MSG_CHECKING([arch option])
 AS_CASE([$1:"$host_cpu"],
     [arm64:arm*],        [ARCH_FLAG=-m64],
-    [arm*:arm*],         [ARCH_FLAG=-m32],
     [x86_64:[i[3-6]86]], [ARCH_FLAG=-m64],
     [x64:x86_64],        [],
     [[i[3-6]86]:x86_64], [ARCH_FLAG=-m32],
